lib/src/facts/aix/serial_number_resolver.cc

```diff
@@ -0,0 +1,34 @@
+#include <internal/facts/aix/serial_number_resolver.hpp>
+#include <internal/util/aix/odm.hpp>
+#include <facter/facts/collection.hpp>
+#include <facter/facts/fact.hpp>
+#include <facter/facts/scalar_value.hpp>
+
+#include <sys/cfgodm.h>
+
+using namespace std;
+using namespace facter::util::aix;
+
+namespace facter { namespace facts { namespace aix {
+
+    serial_number_resolver::serial_number_resolver() :
+        resolver(
+            "AIX serial number",
+            {
+                fact::serial_number,
+             })
+    {
+    }
+
+    void serial_number_resolver::resolve(collection& facts) {
+        auto cu_at_query = odm_class<CuAt>::open("CuAt").query("name=sys0 and attribute=systemid");
+        auto result = *cu_at_query.begin();
+
+        // the ODM returns a string of the form "IBM,XXSERIAL#". We
+        // need to strip the "IBM,XX" from the start ofthis, and keep
+        // only the 7 actual serial number bytes.
```

Comment:
Agreed, I'd prefer us not to crash if the returned value is not what is expected.